### DISCUSSION TITLE
replit免费部署ChatSydney

### DISCUSSION THREAD

**Discussion by renqabs** [viopsa233/AI-testers-Forum#79] [技术交流]
1 准备github账号一个，登录replit https://replit.com/
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/6ea3f2db-26d0-4697-9495-a6b901bedb5c)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/7eda897b-da32-480b-9039-eaf654f8beec)

2 点击Create Repl，Template选择python，再点击Import from GitHub，填入GitHub URL（可以先填我的https://github.com/renqabs/ChatSydney-react-img.git），Language选择python，最后点击Import from GitHub
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/90e98e64-827b-4844-9ebc-488ab5920f2a)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/e273eaba-d599-477d-b5de-cae79abf3e12)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/03ac5fe6-aa02-4219-8e1a-1524f47d462c)

3 设置Configure the Run Command，填入python3 main.py --host 0.0.0.0:65432，点击Done进行保存
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/fe8430ff-9199-4cae-86c8-6d61f12c2ba1)
~~4 安装依赖库，选择shell，输入命令pip install -r requirements.txt，然后执行（有失败的话，就多执行几次命令）~~

5 点击 RUN，等待程序运行成功即可，成功后会自动打开一个web小窗口，窗口内有web访问链接，也可以点击右侧箭头，直接在新标签页并打开链接
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/9d6f5bde-da7f-4d97-9c48-8c104f8ae855)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/55f1e47f-7f62-48f7-93bf-fe2016b11ac6)

**Comment by kennykliu** [comment]
This repository could not be found.这怎么办

**Reply by renqabs** [reply]
是不是填错了github仓库地址，要填https://github.com/renqabs/ChatSydney-react-img.git

**Comment by kennykliu** [comment]
![Uploading Screenshot_2023-07-05-17-29-50-81_76eaced432273cd65da1ec13409568ff.jpg…]()

**Comment by aiss693** [comment]
File ".gitattributes", line 2
    * text=auto
    ^^^^^^
SyntaxError: starred assignment target must be in a list or tuple
 
run之后显示这个😭

**Reply by renqabs** [reply]
有些错误可以忽略掉，不影响正常使用，你直接访问下url，看看正常不

**Comment by kennykliu** [comment]
成功了谢谢大佬🤗

**Comment by istudylove** [comment]
写个shell脚本可以一键布署，还可以自动更新新版

**Comment by xuyufenfei** [comment]
![Screenshot_2023-07-10-00-49-06-567_com replit app](https://github.com/viopsa233/AI-testers-Forum/assets/133896005/820925f3-77ff-454e-9bff-e61a24440c9e)

**Reply by renqabs** [reply]
应该是public/index.html里一行代码的问题（改为wss应该就没问题了），你的可能跟我的不太一样，可以搜索一下
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130155002/7343ba2e-3a52-4f54-9b07-49d5adc7724b)

**Comment by twsec-pro** [comment]
突然报错：
"Traceback (most recent call last):\n  File \"/home/runner/ChatSydney-react-img/venv/lib/python3.10/site-packages/websockets/legacy/client.py\", line 662, in __await_impl__\n    await protocol.handshake(\n  File \"/home/runner/ChatSydney-react-img/venv/lib/python3.10/site-packages/websockets/legacy/client.py\", line 329, in handshake\n    raise InvalidStatusCode(status_code, response_headers)\nwebsockets.exceptions.InvalidStatusCode: server rejected WebSocket connection: HTTP 200\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/runner/ChatSydney-react-img/venv/lib/python3.10/site-packages/websockets/legacy/client.py\", line 655, in __await_impl_timeout__\n    return await self.__await_impl__()\n  File \"/home/runner/ChatSydney-react-img/venv/lib/python3.10/site-packages/websockets/legacy/client.py\", line 676, in __await_impl__\n    await protocol.wait_closed()\nasyncio.exceptions.CancelledError\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/runner/ChatSydney-react-img/main.py\", line 25, in sydney_process_message\n    async for _, response in chatbot.ask_stream(prompt=user_message, conversation_style=\"creative\",raw=True,\n  File \"/home/runner/ChatSydney-react-img/SydneyGPT/SydneyGPT.py\", line 30, in ask_stream\n    async for key, value in super().ask_stream(*args, **kwargs):\n  File \"/home/runner/ChatSydney-react-img/EdgeGPT/EdgeGPT.py\", line 186, in ask_stream\n    async for response in self.chat_hub.ask_stream(\n  File \"/home/runner/ChatSydney-react-img/SydneyGPT/SydneyGPT.py\", line 55, in ask_stream\n    async for key, ...

**Reply by renqabs** [reply]
云平台好像突然都不行了，我本地部署是可以的，不过有的人本地部署也不行，估计是魔法的问题，我暂时也没找到原因，猜测可能是MS封了一些IP吧

**Reply by xuyufenfei** [reply]
那大佬，是否可以通过设置代理来绕过replit的ip呢，求大佬研究一下

**Comment by Cola113** [comment]
您好，尝试对话出现以下报错:

`"Traceback (most recent call last):\n  File \"/home/runner/imgtest/EdgeGPT/conversation.py\", line 154, in create\n    self.struct = response.json()\n  File \"/home/runner/imgtest/venv/lib/python3.10/site-packages/httpx/_models.py\", line 756, in json\n    return jsonlib.loads(self.text, **kwargs)\n  File \"/nix/store/hd4cc9rh83j291r5539hkf6qd8lgiikb-python3-3.10.8/lib/python3.10/json/__init__.py\", line 346, in loads\n    return _default_decoder.decode(s)\n  File \"/nix/store/hd4cc9rh83j291r5539hkf6qd8lgiikb-python3-3.10.8/lib/python3.10/json/decoder.py\", line 337, in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n  File \"/nix/store/hd4cc9rh83j291r5539hkf6qd8lgiikb-python3-3.10.8/lib/python3.10/json/decoder.py\", line 355, in raw_decode\n    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\njson.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/runner/imgtest/main.py\", line 24, in sydney_process_message\n    chatbot = await Chatbot.create(cookies=cookies, proxy=args.proxy, imageInput=imageInput)\n  File \"/home/runner/imgtest/SydneyGPT/SydneyGPT.py\", line 20, in create\n    obj = await EdgeChatBot.create(*args, **kwargs)\n  File \"/home/runner/imgtest/EdgeGPT/EdgeGPT.py\", line 43, in create\n    await Conversation.create(self.proxy, cookies=cookies, imageInput=imageInput),\n  File \"/home/runner/imgtest/EdgeGPT/conversation.py\", line 157, in create\n    raise Exception(\nException: Authentication failed. You have not been accepted into the beta.\n"`
